pkg/filesystem/path/windows_parser.go

```diff
@@ -0,0 +1,152 @@
+package path
+
+import (
+	"strings"
+
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+)
+
+type windowsParser struct {
+	path string
+}
+
+type windowsRelativeParser struct {
+	path string
+}
+
+// NewWindowsParser creates a Parser for Windows paths that can be used in Resolve.
+func NewWindowsParser(path string) (Parser, error) {
+	// Windows paths are generally passed to system calls that
+	// accept C strings. There is no way these can accept null
+	// bytes.
+	if strings.ContainsRune(path, '\x00') {
+		return nil, status.Error(codes.InvalidArgument, "Path contains a null byte")
+	}
```

Comment:
What are your thoughts on using
```
if strings.ContainsFunc(component, unicode.IsControl) || strings.ContainsAny(component, "<>:\"/\\|?*") {
```
here as well? Probably best if we put that in a helper function. Though for paths `:` *are* allowed. It's just that they are not allowed for filenames. Maybe it's best to remove this check entirely from `NewWindowsParser()` entirely, and only perform it as part of `Parse*()`?

Reply:
I am fine with the parse then validate and use split. And then only basic verification is needed here, just like in the UNIX case. That strategy is clear for me now that I have worked with it. But it may need to be articulated further in the parser documentation.